### PR TITLE
Remove duplicated typedef in example when HIP

### PR DESCRIPTION
**Pull request by takagi**
Rel #4132, #4484.

hipRTC fails with the conflicting typedef.

```
cupy.cuda.compiler.CompileException: /tmp/comgr-17edf0/input/CompileSource:5358:32: error: typedef redefinition with different types ('unsigned long long' vs '__uint64_t' (aka 'unsigned long'))
    typedef unsigned long long uint64_t;
                               ^
/usr/include/x86_64-linux-gnu/bits/stdint-uintn.h:27:20: note: previous definition is here
typedef __uint64_t uint64_t;
                   ^
```

### CODE REVIEW

**Comment by takagi** [comment]
pfnCI, test this please.

**Comment by takagi** [comment]
pfnCI, test this please.

**Review by asi1024 — APPROVED** [review]

**Comment by asi1024** [comment]
LGTM.

File: examples/finance/monte_carlo.py [hunk 1]
```diff
@@ -43,7 +43,9 @@
     call = discount_factor * call_sum / n_samples;
     ''',
     preamble='''
+    #ifndef __HIPCC__
```
Comment (leofang) [inline examples/finance/monte_carlo.py:46]: FYI in other places we use `__HIP_DEVICE_COMPILE__`, but the effect should be the same, so just a matter of style choice here.

**Pull request merged**